java/org/contikios/cooja/contikimote/interfaces/ContikiRS232.java

```diff
@@ -147,7 +147,7 @@ public Mote getMote() {
   }
 
   private TimeEvent pendingBytesEvent = null;
-  private final Vector<Byte> pendingBytes = new Vector<>();
+  private final ArrayDeque<Byte> pendingBytes = new ArrayDeque<>();
```

Comment:
Although `ArrayDeque` is a more efficient choice than `Vector`, it is not thread-safe and access needs to be protected as `pendingBytes` can be used from multiple threads (for example, when serial-port is connected to external TCP stream).

Reply:
I look into this further, and the current plugins calling this method seem to do so by first switching to the simulation thread. The switch to `ArrayDeque` should be safe if we assume all will do so. It will then be unnecessary for this method to switch to the simulation thread itself, but that is code cleanup that can be addressed in another PR.